File: DSA/STACK/STACK USING LINKEDLIST/creation.c
// stack creation with linked list along with the operations like pop,push,isempty,isfull and traversal...
#include<stdio.h>
#include<stdlib.h>
struct node
{
    int data;
    struct node * next;
}*top=NULL;
void traversal()
{
    struct node * temp;
    temp=top;
    int i = 1;
    while(temp!=NULL)
    {
        printf("Element on position %d is %d:\n",i,temp->data);
        temp = temp->next;
        i++;
    }
}
int isempty()
{
    if(top==NULL)
    {
        return 1;
    }
    else
    {
        return 0;
    }
}
int isfull()
{
    struct node * temp = (struct node*)malloc(sizeof(struct node));
    if(temp==NULL)
    {
        return 1;
    }
    else
    {
        return 0;
    }
}
void push(int element)
{
    if(isfull())
    {
        printf("Stack overflow\n");
    }
    else
    {
        struct node * temp = (struct node*)malloc(sizeof(struct node));
        temp->data = element;
        temp->next = top;
        top = temp;
        printf("Element %d pushed sucessfully\n",element);
    }
}
int pop()
{
    if(isempty())
    {
        printf("Stack underflow\n");
    }
    else
    {
        struct node * temp = malloc(sizeof(struct node));
        temp->data = top->data;
        top = top->next;
        return temp->data;
        free(temp);
    }
}
void main()
{
    push(1);
    push(2);
    push(3);
    push(4);
   printf("Element %d popped sucessfully\n",pop());
   traversal();
}

File: Java/JAVA 3.0/LinkedList/creation.java
package LinkedList;
class node{
    int data;
    node next;
    node(int data)
    {
        this.data = data;
    }
}

public class creation {

    static void traverse(node head)
    {
        if(head !=null)
        {
            traverse(head.next);
            System.out.println(head.data);
        }else{
            return;
        }
    }


    public static void main(String[] args) {
        node a = new node(5);
        node b = new node(7);
        node c = new node(3);
        node d = new node(9);

        a.next = b;
        b.next = c;
        c.next = d;

        traverse(a);

    }
}
